gst/nnstreamer/include/nnstreamer_plugin_api_trainer.h

```diff
@@ -0,0 +1,110 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * NNStreamer API for Tensor_Trainer Sub-Plugins
+ * Copyright (C) 2022 Hyunil Park <hyunil46.park@samsung.com>
+ */
+/**
+ * @file  nnstreamer_plugin_api_filter.h
+ * @date  1 Dec 2022
+ * @brief Mandatory APIs for NNStreamer Trainer sub-plugins (No External Dependencies)
+ * @see https://github.com/nnstreamer/nnstreamer
+ * @author  Hyunil Park <hyunil46.park@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+#ifndef __NNS_PLUGIN_API_TRAINER_H__
+#define __NNS_PLUGIN_API_TRAINER_H__
+
+#include "tensor_typedef.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief GstTensorTrainer's properties for neural network framework (internal data structure)
+ *
+ * Internal data of GstTensorTrainer required by tensor_trainer's custom subplugin.
+ */
+typedef struct _GstTensorTrainerProperties
+{
+  GstTensorsInfo input_meta;      /**< configured input tensor info */
+  const char *model_config;     /* The configuration file path for creating model */
+  const char *model_save_path;      /* The file path to save the model */
+  unsigned int num_inputs;      /* The number of input lists, the input is where framework receive the features to train the model, num_inputs indicates how many inputs there are. */
+  unsigned int num_labels;      /* The number of label lists, the label is where framework receive the class to train the model, num_labels indicates how many labels there are. */
+  unsigned int num_train_samples;       /* The number of train sample used to train the model. */
+  unsigned int num_valid_samples;       /* The number of valid sample used to train the model. */
+} GstTensorTrainerProperties;
+
+/**
+ * @brief tensor_trainer Subplugin definition
+ *
+ * Common callback parameters:
+ * prop Trainer properties. Read Only.
+ * private_data Subplugin's private data. Set this (*private_data = XXX) if you want to change trainer->private_data.
+ */
+typedef struct _GstTensorTrainerFramework
+{
+  /* The name of the neural network, searchable by 'framework' property */
+  char *name;
```

Comment:
Please follow what GstTensorFilterFramework v1 did, not v0.

- Separation of properties: GstTensorFilterFrameworkInfo
- Separation of event handling: GstTensorFilterFrameworkEventData

This allows higher flexibility: allowing different configuations in each instance.

Reply:
Okay, Thank you.
I will put what you said while adding features.

Reply:
Thank you!!